ExternalModule.php

```diff
@@ -18,9 +19,38 @@ class ExternalModule extends AbstractExternalModule {
      * @inheritdoc
      */
     function redcap_data_entry_form_top($project_id, $record, $instrument, $event_id, $group_id) {
-        $this->includeJs('js/complexFieldValidationHelper.js');
+        $this->includeJs('js/checkRanges.js');
+
+        global $Proj;
+
+        foreach (array_keys($Proj->forms[$_GET['page']]['fields']) as $field_name) {
+            $field_info = $Proj->metadata[$field_name];
+            
+            if (!$display_mode = Form::getValueInActionTag($field_info['misc'], $name)) {
+                continue;
+            }
+
+            // Split the action value into an array of arrays
+            $field_tag_values = explode(",", $display_mode);
+            for($i = 0; $i < sizeof($field_tag_values); $i++){
+                $field_tag_values[$i] = explode("-", $field_tag_values[$i]);
+            }
+
+            $this->sendVarToJS('complex_field_validation_tag_values', $field_tag_values);
```

Comment:
push field_tag_values to an array indexed by the field_name.  move the call to sendVarToJS outside the loop.